### PR TITLE
implement new "humanized" number formatter

### PR DESCRIPTION
**Pull request by bcolloran**
This PR replaces the number formatting affordances in the dashboard with a new one based on some recent experimentation.

Steps to get this merged:
- [x] fix any lint issues
- [x] clean up code & add comments
- [ ] deprecate existing humanize functions
- [x] go through design iterations

### CODE REVIEW

File: web-local/package.json [hunk 1]
```diff
@@ -29,6 +29,7 @@
     "@babel/preset-env": "^7.16.11",
     "@playwright/test": "^1.28.1",
     "@rollup/plugin-typescript": "^8.3.1",
+    "@stdlib/stdlib": "^0.0.96",
```
Comment (hamilton) [inline web-local/package.json:32]: w
```suggestion
```

File: web-local/package.json [hunk 2]
```diff
@@ -58,6 +59,7 @@
     "should": "^13.2.3",
     "sinon": "^15.0.0",
     "supertest": "^6.2.4",
+    "svelte-awesome-color-picker": "^2.4.1",
```
Comment (hamilton) [inline web-local/package.json:62]: Let's remove this dependency from the branch. I think the `<input type="color" />` should be sufficient
```suggestion
    "svelte-awesome-color-picker": "^2.4.1",
```

File: web-local/package.json [hunk 3]
```diff
@@ -29,6 +29,8 @@
     "@babel/preset-env": "^7.16.11",
     "@playwright/test": "^1.28.1",
     "@rollup/plugin-typescript": "^8.3.1",
+    "@stdlib/random-base": "^0.0.6",
+    "@stdlib/random-shuffle": "^0.0.7",
```
Comment (bcolloran) [inline web-local/package.json:33]: @hamilton I changed this from all of `stdlib` to just the two submodules that are actually being used in the dev route, which is much much slimmer. I can remove it if you want, but it's very useful to be able to draw samples from a seedable RNG for this kind of prototyping, and I don't think including adding just these submodules as dev dependencies has any substantial downsides at this point-- it adds a couple seconds perhaps to `npm install`, but should not impact end user experience. But let me know me know if you want it out, I can come up with a workaround / roll my own seedable RNG that's good enough for this application.

**Comment by bcolloran** [comment]
Note that main currently has a "decimal" option in the number formatting dropdown:

image

I have not removed that option in this branch because
1.  I'm not sure what the design intent of the decimal option was 
2. No matter the intent, I think the option may be obsolete given the new formatter
3. The ts type `NicelyFormattedTypes` that controls this option is sprinkled all over the place, and I didn't want to risk breaking something upstream without @djbarnwal 's input and review

For now, if that option is selected, the new default humanizer is used at the last mile, but I haven't touched any of the upstream code that handles `NicelyFormattedTypes`

Longer term, `NicelyFormattedTypes` should likely be deprecated, but that's a bigger project that can happen later, possibly when the options we want to give to users wrt formatting and number semantics are understood better.

**Review by hamilton — APPROVED** [review]

**Pull request merged**